starknet_py/tests/unit/signer/test_ledger_signer.py

```diff
@@ -0,0 +1,189 @@
+import pytest
+
+from starknet_py.common import create_sierra_compiled_contract
+from starknet_py.constants import EIP_2645_PATH_LENGTH
+from starknet_py.contract import Contract
+from starknet_py.hash.address import compute_address
+from starknet_py.hash.selector import get_selector_from_name
+from starknet_py.net.account.account import Account
+from starknet_py.net.client_models import Call
+from starknet_py.net.full_node_client import FullNodeClient
+from starknet_py.net.models import DeclareV3, DeployAccountV1, InvokeV1, StarknetChainId
+from starknet_py.net.signer.ledger_signer import LedgerSigner
+from starknet_py.tests.e2e.fixtures.accounts import mint_token_on_devnet
+from starknet_py.tests.e2e.fixtures.constants import (
+    ETH_FEE_CONTRACT_ADDRESS,
+    MAX_RESOURCE_BOUNDS,
+)
+from starknet_py.tests.e2e.fixtures.misc import load_contract
+
+
+def test_init_with_invalid_derivation_path():
+    with pytest.raises(ValueError, match="Empty derivation path"):
+        LedgerSigner(derivation_path_str="", chain_id=StarknetChainId.SEPOLIA)
+
+    with pytest.raises(
+        ValueError, match=rf"Derivation path is not {EIP_2645_PATH_LENGTH}-level long"
+    ):
+        LedgerSigner(
+            derivation_path_str="m/2645'/1195502025'/1470455285'/0'/0'/0/0",
+            chain_id=StarknetChainId.SEPOLIA,
+        )
+
+    with pytest.raises(
+        ValueError, match=r"Derivation path is not prefixed with m/2645."
+    ):
+        LedgerSigner(
+            derivation_path_str="m/1234'/1195502025'/1470455285'/0'/0'/0",
+            chain_id=StarknetChainId.SEPOLIA,
+        )
+
+
+compiled_contract = load_contract("HelloStarknet")["sierra"]
+sierra_contract_class = create_sierra_compiled_contract(compiled_contract)
+
+
+@pytest.mark.parametrize(
+    "transaction",
+    [
+        InvokeV1(
+            sender_address=0x1,
+            calldata=[1, 2, 3],
+            max_fee=10000,
+            signature=[],
+            nonce=23,
+            version=1,
+        ),
+        DeployAccountV1(
+            class_hash=0x1,
+            contract_address_salt=0x2,
+            constructor_calldata=[1, 2, 3, 4],
+            max_fee=10000,
+            signature=[],
+            nonce=23,
+            version=1,
+        ),
+        DeclareV3(
+            contract_class=sierra_contract_class,
```

Comment:
I guess we can use a mock object here, perhaps even for entire transactions. It is possible to achieve for example by `Mock` library from `unittest.mock`.

Reply:
Maybe let's address it in separate PR (dedicated for test improvements), as it also applies to the same test for stark curve signer